applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp

```diff
@@ -77,7 +77,7 @@ MCPlasticFlowRule::~MCPlasticFlowRule()
 {
 }
 
-void MCPlasticFlowRule::InitializeMaterial(YieldCriterionPointer pYieldCriterionPointer, HardeningLawPointer pHardeningPointer, const Properties& rProp)
+void MCPlasticFlowRule::InitializeMaterial(YieldCriterionPointer& pYieldCriterionPointer, HardeningLawPointer& pHardeningPointer, const Properties& rProp)
```

Comment:
So in long term it would be better to substitute it with normal pointer (or use template as we discuss it)